src/Configuration/Configuration.php

```diff
@@ -34,8 +49,31 @@ public function bind(Configuration $parent)
         $this->parent = $parent;
     }
 
-    public function set(string $name, $value)
+    public function set(string $name, $value, $callback = null)
     {
+        if (is_callable($callback)) {
+            $this->validations[$name] = $callback;
+        }
+        if (array_key_exists($name, $this->validations) 
+            && !call_user_func($this->validations[$name], $value)
+        ) {
+            throw new ConfigurationException("Config option \"$name\" has an invalid value.");
+        }
+        $closestDistance = 4;
+        foreach (self::PREDEFINED_PARAMS as $predefinedParam) {
+            if ($name === $predefinedParam) {
+                unset($closestParam);
+                break;
+            }
+            $distance = levenshtein($name, $predefinedParam);
+            if ($distance < $closestDistance) {
+                $closestDistance = $distance;
+                $closestParam = $predefinedParam;
+            }
+        }
+        if (isset($closestParam)) {
+            print("<fg=yellow>Warning:</> \"$name\" not found in parameters. Did you mean \"$closestParam\"?\n");
```

Comment:
Use warning func.

Reply:
I've tried to use it, but it asks for a Context and the function is called directly from the `set` function in Configuration file. Any tip here?

Reply:
I think we can check `Context::has()` in warning func and output message without host prefix.

Reply:
I updated it as well. I hope it works.